### PR TITLE
Add analyzer assembly info printing

### PR DESCRIPTION
**Pull request by 333fred**
It would be useful to be able to quickly dump detailed information about analyzer versions from complogs, so we can immediately see, for example, what git commit of the razor compiler was used. This adds just that; we look for the assembly name and AssemblyInformationalVersionAttribute for each analyzer and record that information in the log for printing. We also do this for all references, since it goes through the same code path, but no display of that information is currently done.

### CODE REVIEW

**Comment by 333fred** [comment]
An example output from a local blazor test app:
```
[I] at 17:18:13 in /tmp/test via .NET 9.0.100-preview.7.24379.15 took 3s 
➜ ~/git/complog/artifacts/bin/Basic.CompilerLog/debug_net8.0/Basic.CompilerLog print -a
Using /tmp/test/msbuild.binlog
Projects
        test.csproj (net9.0)
Analyzers
        Name: Microsoft.AspNetCore.Analyzers
        Informational Version: 9.0.0-preview.7.24379.2+41a338d8befba633f5349a0344842ae1a9211817
        File Path: /home/fred/.dotnet/sdk/9.0.100-preview.7.24379.15/Sdks/Microsoft.NET.Sdk.Web/analyzers/cs/Microsoft.AspNetCore.Analyzers.dll
        Name: Microsoft.AspNetCore.Mvc.Analyzers
        Informational Version: 9.0.0-preview.7.24379.2+41a338d8befba633f5349a0344842ae1a9211817
        File Path: /home/fred/.dotnet/sdk/9.0.100-preview.7.24379.15/Sdks/Microsoft.NET.Sdk.Web/analyzers/cs/Microsoft.AspNetCore.Mvc.Analyzers.dll
        Name: Microsoft.CodeAnalysis.CSharp.NetAnalyzers
        Informational Version: 9.0.0-preview.24324.1+43709af7570da7140fb3e9a5237f55ffb24677e7
        File Path: /home/fred/.dotnet/sdk/9.0.100-preview.7.24379.15/Sdks/Microsoft.NET.Sdk/targets/../analyzers/Microsoft.CodeAnalysis.CSharp.NetAnalyzers.dll
        Name: Microsoft.CodeAnalysis.NetAnalyzers
        Informational Version: 9.0.0-preview.24324.1+43709af7570da7140fb3e9a5237f55ffb24677e7
        File Path: /home/fred/.dotnet/sdk/9.0.100-preview.7.24379.15/Sdks/Microsoft.NET.Sdk/targets/../analyzers/Microsoft.CodeAnalysis.NetAnalyzers.dll
        Name: Microsoft.Interop.ComInterfaceGenerator
        Informational Version: 9.0.0-preview.7.24376.15+bdcf3ef75e11d22e4973bf37f2d5b660b73aa266
        File Path: /home/fred/.dotnet/packs/Microsoft.NETCore.App.Ref/9.0.0-preview.7.24376.15/analyzers/dotnet/cs/Microsoft.Interop.ComInterfaceGenerator.dll
        Name: Microsoft.Interop.JavaScript.JSImportGenerator
        Informational Version: 9.0.0-preview.7.24376.15+bdcf3ef75e11d22e4973bf37f2d5b660b73aa266
        File Path: /home/fred/.dotnet/packs/Microsoft.NETCore.App.Ref/9.0.0-preview.7.24376.15/analyzers/dotnet/cs/Microsoft.Interop.JavaScript.JSImportGenerator.dll
        Name: Microsoft.Interop.LibraryImportGenerator
        Informational Version: 9.0.0-preview.7.24376.15+bdcf3ef75e11d22e4973bf37f2d5b660b73aa266
        File Path: /home/fred/.dotnet/packs/Microsoft.NETCore.App.Ref/9.0.0-preview.7.24376.15/analyzers/dotnet/cs/Microsoft.Interop.LibraryImportGenerator.dll
        Name: Microsoft.Interop.SourceGeneration
        Informational Version: 9.0.0-preview.7.24376.15+bdcf3ef75e11d22e4973bf37f2d5b660b73aa266
        File Path: /home/fred/.dotnet/packs/Microsoft.NETCore.App.Ref/9.0.0-preview.7.24376.15/analyzers/dotnet/cs/Microsoft.Interop.SourceGeneration.dll
        Name: System.Text.Json.SourceGeneration
        Informational Version: 9.0.0-preview.7.24376.15+bdcf3ef75e11d22e4973bf37f2d5b660b73aa266
        File Path: /home/fred/.dotnet/packs/Microsoft.NETCore.App.Ref/9.0.0-preview.7.24376.15/analyzers/dotnet/cs/System.Text.Json.SourceGeneration.dll
        Name: System.Text.RegularExpressions.Generator
        Informational Version: 9.0.0-preview.7.24376.15+bdcf3ef75e11d22e4973bf37f2d5b660b73aa266
        File Path: /home/fred/.dotnet/packs/Microsoft.NETCore.App.Ref/9.0.0-preview.7.24376.15/analyzers/dotnet/cs/System.Text.RegularExpressions.Generator.dll
        Name: Microsoft.AspNetCore.App.Analyzers
        Informational Version: 9.0.0-preview.7.24379.2+41a338d8befba633f5349a0344842ae1a9211817
        File Path: /home/fred/.dotnet/packs/Microsoft.AspNetCore.App.Ref/9.0.0-preview.7.24379.2/analyzers/dotnet/cs/Microsoft.AspNetCore.App.Analyzers.dll
        Name: Microsoft.AspNetCore.App.CodeFixes
        Informational Version: 9.0.0-preview.7.24379.2+41a338d8befba633f5349a0344842ae1a9211817
        File Path: /home/fred/.dotnet/packs/Microsoft.AspNetCore.App.Ref/9.0.0-preview.7.24379.2/analyzers/dotnet/cs/Microsoft.AspNetCore.App.CodeFixes.dll
        Name: Microsoft.AspNetCore.Components.Analyzers
        Informational Version: 9.0.0-preview.7.24379.2+41a338d8befba633f5349a0344842ae1a9211817
        File Path: /home/fred/.dotnet/packs/Microsoft.AspNetCore.App.Ref/9.0.0-preview.7.24379.2/analyzers/dotnet/cs/Microsoft.AspNetCore.Components.Analyzers.dll
        Name: Microsoft.Extensions.Logging.Generators
        Informational Version: 9.0.0-preview.7.24376.15+bdcf3ef75e11d22e4973bf37f2d5b660b73aa266
        File Path: /home/fred/.dotnet/packs/Microsoft.AspNetCore.App.Ref/9.0.0-preview.7.24379.2/analyzers/dotnet/roslyn4.4/cs/Microsoft.Extensions.Logging.Generators.dll
        Name: Microsoft.Extensions.Options.SourceGeneration
        Informational Version: 9.0.0-preview.7.24376.15+bdcf3ef75e11d22e4973bf37f2d5b660b73aa266
        File Path: /home/fred/.dotnet/packs/Microsoft.AspNetCore.App.Ref/9.0.0-preview.7.24379.2/analyzers/dotnet/roslyn4.4/cs/Microsoft.Extensions.Options.SourceGeneration.dll
        Name: Microsoft.AspNetCore.Razor.Utilities.Shared
        Informational Version: 9.0.0-preview.24376.2+6cf62038c2c42e72901b9fb759f1be7b2779fd3e
        File Path: /home/fred/.dotnet/sdk/9.0.100-preview.7.24379.15/Sdks/Microsoft.NET.Sdk.Razor/targets/../source-generators/Microsoft.AspNetCore.Razor.Utilities.Shared.dll
        Name: Microsoft.CodeAnalysis.Razor.Compiler
        Informational Version: 9.0.0-preview.24376.2+6cf62038c2c42e72901b9fb759f1be7b2779fd3e
        File Path: /home/fred/.dotnet/sdk/9.0.100-preview.7.24379.15/Sdks/Microsoft.NET.Sdk.Razor/targets/../source-generators/Microsoft.CodeAnalysis.Razor.Compiler.dll
        Name: Microsoft.Extensions.ObjectPool
        Informational Version: 8.0.0+3f1acb59718cadf111a0a796681e3d3509bb3381
        File Path: /home/fred/.dotnet/sdk/9.0.100-preview.7.24379.15/Sdks/Microsoft.NET.Sdk.Razor/targets/../source-generators/Microsoft.Extensions.ObjectPool.dll
        Name: System.Collections.Immutable
        Informational Version: 8.0.0+5535e31a712343a63f5d7d796cd874e563e5ac14
        File Path: /home/fred/.dotnet/sdk/9.0.100-preview.7.24379.15/Sdks/Microsoft.NET.Sdk.Razor/targets/../source-generators/System.Collections.Immutable.dll
```

I've tested both from binlogs and a complog.

File: src/Basic.CompilerLog/Program.cs [hunk 1]
```diff
@@ -172,6 +174,18 @@ int RunPrint(IEnumerable<string> args)
         foreach (var compilerCall in compilerCalls)
         {
             WriteLine($"\t{compilerCall.GetDiagnosticName()}");
+
+            if (analyzers)
+            {
+                WriteLine("Analyzers");
```
Comment (jaredpar) [inline src/Basic.CompilerLog/Program.cs:180]: Shouldn't this be indented? In the case of multiple projects this indentation will be a bit jumpy.

File: src/Basic.CompilerLog/Program.cs [hunk 2]
```diff
@@ -75,7 +75,7 @@ int RunCreate(IEnumerable<string> args)
         var convertResult = CompilerLogUtil.TryConvertBinaryLog(binlogFilePath, complogFilePath, options.FilterCompilerCalls);
         foreach (var diagnostic in convertResult.Diagnostics)
         {
-           WriteLine(diagnostic);
+            WriteLine(diagnostic);
```
Comment (jaredpar) [inline src/Basic.CompilerLog/Program.cs:78]: Did you just notice the bad formatting or did you do something in VS code to make it format the docs?

File: src/Basic.CompilerLog.Util/ReferenceData.cs [hunk 3]
```diff
@@ -14,10 +14,12 @@ public sealed class ReferenceData(string filePath, Guid mvid, byte[] imageBytes)
     /// </remarks>
     public string FilePath { get; } = filePath;
     public Guid Mvid { get; } = mvid;
+    public string? AssemblyName { get; } = assemblyName;
+    public string? AssemblyInformationalVersion { get; } = assemblyInformationalVersion;
```
Comment (jaredpar) [inline src/Basic.CompilerLog.Util/ReferenceData.cs:18]: Where are these used?

File: src/Basic.CompilerLog/Program.cs (hunk 2 above)
Comment (333fred) [reply src/Basic.CompilerLog/Program.cs:78]: Yeah, I hit format document and it found all of these.

File: src/Basic.CompilerLog.Util/ReferenceData.cs (hunk 3 above)
Comment (333fred) [reply src/Basic.CompilerLog.Util/ReferenceData.cs:18]: The loop in `Program` reads these, `ReadAllAnalyzerData` returns these.

File: src/Basic.CompilerLog/Program.cs (hunk 1 above)
Comment (333fred) [reply src/Basic.CompilerLog/Program.cs:180]: Yeah, I missed the `\t` in the compiler call name.

**Review by 333fred — COMMENTED** [review]
@jaredpar addressed feedback.

**Comment by jaredpar** [comment]
Looks ilke u have real test failures

**Comment by 333fred** [comment]
> Looks ilke u have real test failures

Fixed. Names matter, turns out.

**Pull request merged**